src/java_bytecode/java_types.cpp

```diff
@@ -389,6 +414,55 @@ typet java_type_from_string(
 
       return java_reference_type(symbol_type);
     }
+  case '*':
+  {
```

Comment:
Probably these three cases (`'*'`, `'-'`, `'+'`) should be handled by throwing an exception rather than this ugly partial handling.

Reply:
I'd agree with @thk123

Reply:
For the record - I should say it was me who wrote the  "ugly partial handling" :stuck_out_tongue: